shell/egl_window.cc

```diff
@@ -305,37 +289,77 @@ int EglWindow::create_shm_buffer(Display* display,
 void EglWindow::handle_xdg_surface_configure(void* data,
                                              struct xdg_surface* xdg_surface,
                                              uint32_t serial) {
-  auto* w = reinterpret_cast<EglWindow*>(data);
+  (void)data;
   xdg_surface_ack_configure(xdg_surface, serial);
-  w->m_xdg_surface_is_configured = true;
 }
 
-#if HIDPI_CLIENT
-void EglWindow::handle_primary_surface_enter(void* data,
-                                             struct wl_surface* wl_surface,
-                                             struct wl_output* output) {
-  (void)data;
-  (void)wl_surface;
-  (void)output;
-  FML_DLOG(INFO) << "Surface Entered ";
-}
+const struct xdg_surface_listener EglWindow::xdg_surface_listener = {
+    .configure = handle_xdg_surface_configure};
 
-void EglWindow::handle_primary_surface_leave(void* data,
-                                             struct wl_surface* wl_surface,
-                                             struct wl_output* output) {
-  (void)data;
-  (void)wl_surface;
-  (void)output;
-  FML_DLOG(INFO) << "Surface Leave";
+void EglWindow::handle_toplevel_configure(void* data,
+                                          struct xdg_toplevel* toplevel,
+                                          int32_t width,
+                                          int32_t height,
+                                          struct wl_array* states) {
+  (void)toplevel;
+  auto* w = reinterpret_cast<EglWindow*>(data);
+
+  w->m_fullscreen = false;
+  w->m_maximized = false;
+
+  const uint32_t* state;
+  WL_ARRAY_FOR_EACH(state, states, const uint32_t*) {
+    switch (*state) {
+      case XDG_TOPLEVEL_STATE_FULLSCREEN:
+        w->m_fullscreen = true;
+        break;
+      case XDG_TOPLEVEL_STATE_MAXIMIZED:
+        w->m_maximized = true;
+        break;
+      case XDG_TOPLEVEL_STATE_RESIZING:
+        w->m_resize = true;
+        break;
+      case XDG_TOPLEVEL_STATE_ACTIVATED:
+        w->m_activated = true;
+        break;
+    }
+  }
+
+  if (width > 0 && height > 0) {
+    if (!w->m_fullscreen && !w->m_maximized) {
+      w->m_window_size.width = width;
+      w->m_window_size.height = height;
+    }
+    w->m_geometry.width = width;
+    w->m_geometry.height = height;
+
+    /* Initialise damage to full surface, so the padding gets painted */
+    wl_surface_damage(w->m_surface, 0, 0, w->m_geometry.width,
```

Comment:
No need to inflict damage for the surface.

Reply:
removed